src/nyaml/nxdl2nyaml.py

```diff
@@ -749,12 +765,20 @@ def handle_dim_with_all_dim_attr(depth, node, file_out, possible_dim_attrs):
             # index and value attributes of dim elements
             indent = (depth + 1) * DEPTH_SIZE
             # Numpy style for dim
-            value = (
-                ", ".join(dim_index_value)
-                if len(dim_index_value) > 1
-                else f"{dim_index_value[0]},"
-            )
-            file_out.write(f"{indent}dim: ({value})\n")
+            if len(dim_index_value) > 0:
+                if not all(val == "" for val in dim_index_value):
+                    value = (
+                        ", ".join(dim_index_value)
+                        if len(dim_index_value) > 1
+                        else f"{dim_index_value[0]},"
+                    )
+                    file_out.write(f"{indent}dim: ({value})\n")
+                else:
+                    if len(dim_index_value) == 1:
+                        value = "1,"
+                        file_out.write(f"{indent}dim: ({value})\n")
+                    else:
+                        raise NotImplementedError()
```

Comment:
Even, if `dim_index_value` has at least 1 element, the code writing a default value `1`, which loses the information coming through the nxdl file. 
What would be the case if `dim_index_value` more then one element?

Reply:
Indeed, if there was nothing in nxdl.xml specified, we shall not wreite anything to nyaml either.
@mkuehbach please revise!

Reply:
done, refactored